### DISCUSSION TITLE
glad to see someone keep maintaining this wonderful tool finally

### DISCUSSION THREAD

**Discussion by jacobax** [sylikc/jpegview#45] [General]
Ive been using jpegview for a long time as it best suits my needs
Thank you.

**Comment by sylikc** [comment]
Thanks for your support!  There's a few pull requests out and hopefully new features will be added over time!

**Reply by bitsper2nd** [reply]
some heads up. I'm working on a fluent design concept for jpeview on figma. It's rough for now. I'm working on the nav panel and the icon. 

![navpanel](https://user-images.githubusercontent.com/22452474/160467536-4a421b5d-0aab-4061-90e8-b71d6ed9b4ea.png)

![icon](https://user-images.githubusercontent.com/22452474/160467591-4261535e-2bfb-46b5-ad71-d912d6c9b8c3.png)

**Reply by sylikc** [reply]
Oh cool @bitsper2nd !  Yes... there was some PR some time ago with icons and stuff, but I feel like a whole icon set and revamp of interface is needed.  I like the way that new icon set looks!

So coincidentally, I want to set up a website for JPEGView,.... (and I haven't worked on website design in... oh... forever).  I was going to just set up something static, some screenshots, a short blurb, etc... on GitHub pages.

Right now https://jpegview.net just lands on the source page... which... in terms of user friendliness is zero haha.  So, I wanted to have like something that a normal user could land on and care.

So, I could use some help with that too 😁

**Reply by bitsper2nd** [reply]
The site isn't loading for me. But sure I can help out with that. We can take cues from [qpv](https://moduleart.github.io/quick-picture-viewer/) and[ qview](https://interversehq.com/qview/). I will post tomorrow more design concepts.

**Reply by sylikc** [reply]
oh, it's not yet created.  Right now the domain just redirects... http://jpegview.net or http://jpegview.org

I was originally just planning to go with Jekyll https://docs.github.com/en/pages/setting-up-a-github-pages-site-with-jekyll/creating-a-github-pages-site-with-jekyll to do the build.  I was doing research on how to do it and thinking about the layout.  No code was written yet.

I like those reference sites lol.  Web 3.0 is so cool... I really haven't written a website for more than a decade 🙄  I can't find a cache anywhere, but let's just say it was plaintext, and my colors were mismatched 🤣

**Comment by c-sanchez** [comment]
Although I don't really like the all flat style that microsoft, google, and most of them have taken now, I think it's a very good redesign that you are doing, @bitsper2nd

I would even say it is less flat than the original jpegview design itself, so much the better hehe

**Reply by bitsper2nd** [reply]
In retrospect, Google and Microsoft have moved on from flat design like 3 or 4 years ago. What you see in the [images](https://www.theverge.com/2021/9/17/22679132/microsoft-windows-11-new-photos-app-design-features) uses Microsoft [Fluent Design System](https://fluenticons.co/outlined/).

**Comment by c-sanchez** [comment]
Btw @sylikc one thing I have always found a bit awkward/confusing is the positioning of the buttons to rotate the image to the Left or Right.

I mean, it doesn't make sense to me that the left turn button is on the RIGHT side, and the right turn button is on the LEFT side haha.

Think of your favorite graphic editor, be it photoshop, gimp, krita, whatever, and have the redo button on the left and undo on the right, is counterintuitive.

It is a detail that I never paid much attention to, but it is true that more than once when trying to rotate the images with the buttons I did it in the opposite direction to the one I wanted hahaha.
![image](https://i.ibb.co/hWdRf91/rotation.jpg)

I could see it more clearly with the redesign of @bitsper2nd, which follows the same order of the original buttons.

**Reply by bitsper2nd** [reply]
You are right on that assessment. I did followed the OG button orientation. I should fix that on the redesign. Thanks for pointing that out!

**Reply by sylikc** [reply]
That could be a change we could make with the panel... Design a good order, and then I'll change it in code.

You can reorder the buttons the way you think would be a good design language and I'll change the underlying bit.

**Comment by bitsper2nd** [comment]
## Intro

I'm ready for the next feedback round. Here is what I got. I'm using [fluent-design system icons](https://fluenticons.co/outlined/) and microsoft photos as an inspiration for the design. The wallpaper in the images is called [Fractal Mountains](https://assets.system76.com/downloads/wallpapers/System76-Fractal_Mountains-by_Kate_Hazen_of_System76.png) from System76.

![ms-photo](https://user-images.githubusercontent.com/22452474/160693747-93bc5d89-486d-4700-94c1-ea732d63b765.png)

### Panels

- nav panel
![NavPanel-new-short](https://user-images.githubusercontent.com/22452474/160694009-47113acf-caa1-40d6-886d-127d0d2e248e.png)
- popup panel
![PopupPanel-new](https://user-images.githubusercontent.com/22452474/160694037-ba837f7e-7d82-4557-a5ed-6c2ffbe455c2.png)

### Old vs New

![old-nav](https://user-images.githubusercontent.com/22452474/160694496-aca10226-0f44-4d6d-9411-a60ea54edc35.png)
![new-nav](https://user-images.githubusercontent.com/22452474/160694508-e9391f67-8c5a-448c-b32b-bdc1b3b74989.png)

![old-edit](https://user-images.githubusercontent.com/22452474/160694523-aaa17dd4-da72-416f-8ece-c829a07eec0f.png)
![new-edit](https://user-images.githubusercontent.com/22452474/160694540-41f2130c-7e49-477e-8a41-a587a0c3ea70.png)

### Notes
I will change the design tomorrow. Better alignment. Use of the Sego UI font. Add separators between icons. So nothing is final. Everything is subject to change. Here are the [pngs](https://github.com/sylikc/jpegview/files/8374608/JPEGviewNew.zip) for the possible icon.

**Reply by sylikc** [reply]
great work!  I like it... the dialogs might not be as easy to change as the panel.  I think I'd be able to figure it out at some point with some work, but the icons in the panel for sure... and i'm sure another dev might point me int he right direction in terms of how to change the panels..

**Comment by c-sanchez** [comment]
I wonder what the new design would look like with some transparency.
Maybe it's not really necessary, maybe it's better, maybe it's worse with transparency.

Or have an option to use the menus with or without transparency.
I don't know, but I guess it would be nice to see for comparison.

**Reply by bitsper2nd** [reply]
You mean [add mica](https://github.com/minusium/MicaForEveryone) to the program window controls and context menu? I will upload that concept in here this weekend.

**Reply by sylikc** [reply]
Erm... If it's a feature that is in addition to whatever JPEGView does now... i wouldn't count on it being available yet...

JPEGView still uses some older APIs and I think that's good for compatibility reasons... some of the newer calls which support transparency could be added over time, but i'd say later rather than sooner :/

**Comment by bitsper2nd** [comment]
Today's Changelog:
- more refinements using the red square method
- new icon design
- better alignment in the navigation panel and popup
- Using Segoe UI font family included in Windows 10 and Windows 11

![Icon-design](https://user-images.githubusercontent.com/22452474/161093047-72f53321-b451-4978-b3d9-2a51a736d85a.png)
Here are the icons for comparison.

![Icon-size](https://user-images.githubusercontent.com/22452474/161093120-070b1050-8abc-4e01-b97a-508c2f8c7b51.png)
Here are icons in from size 16 to 256. 

![Nav-panel](https://user-images.githubusercontent.com/22452474/161093173-f9041cab-370d-4fe5-a462-478a6674d340.png)
Here is the improved navigation panel.

![Popup-panel](https://user-images.githubusercontent.com/22452474/161093200-f6d1c821-48cd-4040-a6d9-044f781f7d8f.png)
Here is the improved popup.

**Reply by sylikc** [reply]
I definitely like the new modern feel... can't decide which icon gradient looks more visually appealing lol

**Reply by bitsper2nd** [reply]
> I definitely like the new modern feel... can't decide which icon gradient looks more visually appealing lol

Here are the PNGs in case you want a closer [look.](https://github.com/sylikc/jpegview/files/8400150/JPEGview.Redesign.zip)

**Comment by bitsper2nd** [comment]
Tonight's Changelog:

- removed the border from icon
- tested mica in jpegview
- updated the mockups
- updated the icon sizes

![Alt-icon-size](https://user-images.githubusercontent.com/22452474/161655526-29b73f69-52a7-4517-9f63-084032459d1f.png)

[Here is the file](https://github.com/sylikc/jpegview/files/8419478/JPEGview.zip) for the icons. It comes with PNG, SVG, and ICO formats. 

![nav](https://user-images.githubusercontent.com/22452474/161655565-55f8240d-937a-41e1-b636-d3c540b2f311.png)
![popup](https://user-images.githubusercontent.com/22452474/161655567-fbbe4d52-a6f7-4092-bc9f-05a56f092dcf.png)

If anyone else wants to add mica to the title bar like the in the images, they can do so [with this program called Mica for Everyone](https://github.com/minusium/MicaForEveryone).

**Reply by sylikc** [reply]
@bitsper2nd I liked some of this artwork, and while i've been busy the past few weeks and might still be a bit backlogged with dev tasks, did you do any more updates to your graphics and redesign.  I'd be willing to roll it in as is with your permission

**Reply by bitsper2nd** [reply]
You can use them. I can make more if you want. And make more adjustments with your feedback.

**Reply by Bobbsterman** [reply]
I suppose I can always customize it post-install, but I personally wouldn't like a wallpaper. I think one of the defining features of JPEGView is just how minimal it is. Nice work on the modern UI though! I only suggest you make the margins on everything a little smaller. Again, I've always loved how JPEGView's UI doesn't get too in the way.

Alternatively, go big and bold, but leave an option in the settings to switch the "classic theme"? I gotta admit, I've grown fond of its somewhat retro harsh, austere lines!

**Comment by donatasben** [comment]
Just wanted to say huge thanks for keeping this lovely app alive! Been searching for a similar one a long time until I found the old version. It's perfect for me! Love me some open-to-fullscreen, no-GUI, only-keyboard-navigation usage 😆 Wouldn't imagine anything else for my usage 😃 

Very much looking forward to what you will accomplish/update it! Started using Winget recently so very happy to see JPEGView available with it!

Love the new designs!
I am also web designer so could maybe have some input towards the website design 😉 Although I have very limited time to spare, but would still ove to see it materialize!

Keep up the good work everyone!

**Reply by bitsper2nd** [reply]
Maybe we can work something out on Figma. Now that I'm done with the icon/application mockup, I can do a website mockup.

**Comment by ctp9** [comment]
Here to also say Thank You for keeping this great project alive!

**Comment by teckiepc** [comment]
hi all, thanks in advance for any answers to my question and also for resurfacing this wonderful program

my question, i don't know what i did but the bottom panel is missing. i did not change any parameters and moving the mouse to the bottom does not raise the panel either. so am not sue how to reset everything to default

**Reply by sylikc** [reply]
I found sometimes the bottom panel doesn't show if your window size is too small.  Otherwise, if you download a fresh copy with a fresh INI, is it still missing?

**Comment by sylikc** [comment]
@bitsper2nd it's been a year since you did the redesign, and I have been a bit more active lately with some free time.  I might start using your icons and possibly some of those graphical updates.  

I always think about licensing and stuff and honoring the work of others, so I was just going to make sure in writing (_just to make sure_):

**Are you ok with releasing the work above to JPEGView on the GPLv2 license?**

Or let me know what license you are releasing it under, so I can make sure to note it.  FYI: CC 4.0 and CC-BY 4.0 is GPLv2-compatible, CC-BY-SA 4.0 is GPLv3 compatible. [The Creative Commons license is now compatible with the GPL](https://poststatus.com/the-creative-commons-license-is-now-compatible-with-the-gpl/)

Thanks!

**Reply by bitsper2nd** [reply]
I don't mind at all. Feel free to use the license you believe is the best.

**Comment by pureby** [comment]
@bitsper2nd Stunning designs! I love the way you updated the face without changing the spirit. Both for the main icon and toolbar ones. Thank you for the contribution!
